src/hooks/useVoteReason.ts

```diff
@@ -0,0 +1,34 @@
+import { useMemo } from 'react'
+
+import useAuthContext from 'decentraland-gatsby/dist/context/Auth/useAuthContext'
+
+import { ProposalAttributes } from '../entities/Proposal/types'
+
+import useDelegationOnProposal from './useDelegationOnProposal'
+import useVotingPowerOnProposal from './useVotingPowerOnProposal'
+
+const REASON_THRESHOLD = 50000
+
+function useVoteReason(proposal: ProposalAttributes | null) {
+  const [account] = useAuthContext()
+  const { delegationResult, isDelegationResultLoading } = useDelegationOnProposal(proposal, account)
+  const delegators: string[] = useMemo(
+    () => delegationResult.delegatedFrom.map((delegator) => delegator.delegator),
```

Comment:
We are already doing this in ProposalVotingSection. Maybe we could move this to a `getDelegatorsFromDelegationResult` function or something?